openmrs-19/src/main/java/org/motechproject/openmrs19/resource/impl/PatientResourceImpl.java

```diff
@@ -121,22 +123,34 @@ public void deletePatient(Config config, String uuid) {
 
     @Override
     public void updatePatientMotechId(Config config, String patientUuid, String newMotechId) {
-        Identifier patientIdentifier = getPatientIdentifier(config, patientUuid);
-        patientIdentifier.setIdentifier(newMotechId);
+        List<Identifier> patientIdentifiers = getPatientIdentifierList(config, patientUuid);
+        for (Identifier patientIdentifier : patientIdentifiers) {
+            if ("MOTECH Id".equals(patientIdentifier.getIdentifierType().getDisplay())) {
```

Comment:
this identifier name is configurable. We've got a method to get this name. Please use this here